### PR TITLE
test: add support for web3.js to simulation tests

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Add the support for web3.js to simulation tests. 

**Description**

- We need to test some scenarios in future which requires to interact with the EL nodes
- Having custom logic for all interactions will be redundant and error prune 
- Using proven web3.js as ETH RPC client will solve a lot of issues. 

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/cli/test/utils/simulation/Web3Plugins.ts
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/Web3Plugins.ts:1]: I know the simulation rig has file names in PascalCase but not sure that is the right way to go. Can we name this file `web3JsPlugins.ts`

File: packages/cli/test/utils/simulation/Web3Plugins.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+
+declare module "web3" {
+  interface Web3Context {
+    admin: Web3AdminPlugin;
+  }
+}
+
+export function registerWeb3Plugins(web3: Web3): void {
```
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/Web3Plugins.ts]:
```suggestion
export function registerWeb3JsPlugins(web3: Web3): void {
```

File: packages/cli/test/utils/simulation/execution_clients/geth.ts [hunk 2]
```diff
@@ -166,11 +165,8 @@ export const generateGethNode: ExecutionNodeGenerator<ExecutionClient.Geth> = (o
 
   const job = runner.create([{...initJobOptions, children: [{...importJobOptions, children: [startJobOptions]}]}]);
 
-  const provider = new Eth1ProviderWithAdmin(
-    {DEPOSIT_CONTRACT_ADDRESS: ZERO_HASH},
-    // To allow admin_* RPC methods had to add "ethRpcUrl"
-    {providerUrls: [ethRpcPublicUrl, engineRpcPublicUrl], jwtSecretHex: SHARED_JWT_SECRET}
-  );
+  const web3 = new Web3(ethRpcPublicUrl);
```
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/execution_clients/geth.ts]: I think `provider` feels more industry standard in this context.  I tend to like the old way better so its clear what the object does to observers.

```suggestion
  const provider = new Web3(ethRpcPublicUrl);
```

File: packages/cli/test/utils/simulation/execution_clients/nethermind.ts [hunk 3]
```diff
@@ -130,11 +129,8 @@ export const generateNethermindNode: ExecutionNodeGenerator<ExecutionClient.Neth
 
   const job = runner.create([startJobOptions]);
 
-  const provider = new Eth1ProviderWithAdmin(
-    {DEPOSIT_CONTRACT_ADDRESS: ZERO_HASH},
-    // To allow admin_* RPC methods had to add "ethRpcUrl"
-    {providerUrls: [ethRpcPublicUrl, engineRpcPublicUrl], jwtSecretHex: SHARED_JWT_SECRET}
-  );
+  const web3 = new Web3(ethRpcPublicUrl);
```
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/execution_clients/nethermind.ts]: Same as above.  I tend to think `provider` was the correct name.

File: packages/cli/test/utils/simulation/utils/network.ts [hunk 4]
```diff
@@ -45,16 +45,17 @@ export async function connectNewCLNode(newNode: BeaconNode, nodes: BeaconNode[])
...
 
   for (const node of nodes) {
     if (node === newNode) continue;
 
     // Nethermind had a bug in admin_addPeer RPC call
     // https://github.com/NethermindEth/nethermind/issues/4876
-    if (node.provider !== null && node.client !== ExecutionClient.Nethermind && elIdentity) {
-      await node.provider.admin.addPeer(elIdentity.enode);
+    if (node.web3 !== null && node.client !== ExecutionClient.Nethermind && elIdentity) {
+      // `web3.admin` here refers to the Web3 plugin `Web3AdminPlugin`
```
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/utils/network.ts:57]: I like that you added the note here!

File: packages/cli/test/utils/simulation/Web3Plugins.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import {Web3PluginBase, Web3} from "web3";
+
+class Web3AdminPlugin extends Web3PluginBase {
+  pluginNamespace = "admin";
```
Comment (matthewkeil) [inline packages/cli/test/utils/simulation/Web3Plugins.ts]:
```suggestion
  /**
   * The admin plugin as available via the provider object
   * like in the example below.
   *
   * await node.web3.admin.addPeer(elIdentity.enode);
   */
  pluginNamespace = "admin";
```

**Review by matthewkeil — CHANGES_REQUESTED** [review]
Really nice addition! Just a couple comments on naming.

File: packages/cli/test/utils/simulation/Web3Plugins.ts
Comment (nazarhussain) [reply packages/cli/test/utils/simulation/Web3Plugins.ts:1]: > I know the simulation rig has file names in PascalCase but not sure that is the right way to go. Can we name this file `web3JsPlugins.ts`

Let's do the renaming of all files for consistent pattern in different PR.

**Review by matthewkeil — APPROVED** [review]
LGTM!! 🚀

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.18.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.18.0) :tada:

**Pull request merged**